lib/maths/CBoostedTreeImpl.cc

```diff
@@ -1561,10 +1561,8 @@ void CBoostedTreeImpl::computeShapValues(core::CDataFrame& frame, const TProgres
     }
 }
 
-CBoostedTreeImpl::TSizeVec CBoostedTreeImpl::columnsHoldingShapValues() const {
-    TSizeVec result(m_LastShapColumnIndex - m_FirstShapColumnIndex + 1);
-    std::iota(result.begin(), result.end(), m_FirstShapColumnIndex);
-    return result;
+CBoostedTreeImpl::TSizeRange CBoostedTreeImpl::columnsHoldingShapValues() const {
+    return boost::irange<std::size_t>(m_FirstShapColumnIndex, m_LastShapColumnIndex + 1);
```

Comment:
nit: seems cleaner to use typedef in case we change the type again
```suggestion
    return TSizeRange{m_FirstShapColumnIndex, m_LastShapColumnIndex + 1};
```